include/boost/geometry/algorithms/detail/disjoint/multirange_geometry.hpp

```diff
@@ -47,31 +73,77 @@ class unary_disjoint_geometry_to_query_geometry
     Strategy const& m_strategy;
 };
 
-
 template<typename MultiRange, typename ConstantSizeGeometry>
-struct multirange_constant_size_geometry
+struct multirange_constant_size_geometry_with_info
 {
+
+    typedef typename point_type<ConstantSizeGeometry>::type point_type;
+
+    typedef segment_intersection_points<point_type> intersection_return_type;
+
     template <typename Strategy>
-    static inline bool apply(MultiRange const& multirange,
-                             ConstantSizeGeometry const& constant_size_geometry,
-                             Strategy const& strategy)
+    static inline intersection_return_type
+    apply(MultiRange const& multirange,
+          ConstantSizeGeometry const& constant_size_geometry,
+          Strategy const& strategy)
     {
-        typedef unary_disjoint_geometry_to_query_geometry
+        typedef disjoint_geometry_to_query_geometry
             <
                 ConstantSizeGeometry,
                 Strategy,
-                dispatch::disjoint
+                dispatch::disjoint_with_info
                     <
                         typename boost::range_value<MultiRange>::type,
                         ConstantSizeGeometry
                     >
             > unary_predicate_type;
 
-        return detail::check_iterator_range
+        intersection_return_type res = detail::check_iterator_range_with_info
             <
-                unary_predicate_type
+                ConstantSizeGeometry, unary_predicate_type
             >::apply(boost::begin(multirange), boost::end(multirange),
                      unary_predicate_type(constant_size_geometry, strategy));
```

Comment:
Ok, so here is an example where `check_iterator_range_with_info` is used. I think I'd be ok with leaving it as it is for now. Later when we move to C++14 it would be possible to replace it with `std::find_if` with generic lambda. Now it'd still require to implement a predicate elsewhere but instead of `apply` it'd have `operator()`. So it wouldn't be much better.

Reply:
Ok, I agree.